_posts/2020-06-21-one-stop-guide-to-lombok.md

```diff
@@ -0,0 +1,337 @@
+---
+title: "One-stop Guide to Lombok"
+date: 2020-06-21 10:00 +0500
+modified: 2020-06-21 10:15 +0500
+---
+
+# What is it
+Lombok is a Java library which makes Java less chatty whilst making your code more readable. Java is great language but when you use its features you have to write a lot of boilerplate code.
+
+# Why to use Lombok
+
+Lombok helps you with a set of annotations to make your code more readable without losing its functionality. You don’t need to write typical constructors, setters or getters. With the help of Lombok you can easily add logging feature to you class or create fully functional builder.
+
+# Which the most popular features it has
+
+## @Getter and @Setter annotations
+These annotations generate setters and getters for a field respectively. These annotations can be applied as method level or class level.
+
+Lombok annotated code
+
+```java
+@Getter
+@Setter
+public class Person {
+  private String name;
+  private boolean employed;
+}
+```
+
+Generated Java code
+
+```java
+private String name;
+private boolean employed;
+
+public void setName(final String name) {
+  this.name = name;
+}
+
+public void setEmployed(final boolean employed) {
+    this.employed = employed;
+}
+
+public String getName() {
+    return this.name;
+}
+
+public boolean isEmployed() {
+    return this.employed;
+}
+```
+
+## @EqualsAndHashCode annotation
```

Comment:
Instead of just using the annotations in the header, make the headers "active" like: "Generating `equals()` and `hashcode()`" or "Generating a Builder"